rbc/src/main/java/com/library/rbc/service/BookService.java

```diff
@@ -88,6 +98,22 @@ private List<BookStatusDto> convertStringsToBookStatusesDto(List<String> bookSta
         }).toList();
   }
 
-
+  public String uploadImage(MultipartFile image){
+    String homeDirectory = System.getProperty("user.home");
+    Path uploadPath = Paths.get(homeDirectory, "Documents", "images");
+    if (image.getContentType() == null || !image.getContentType().contains("image")) {
+      throw new ContentTypeException("Provided file must be image");
+    }
+    try {
+      if (!Files.exists(uploadPath)) {
+        Files.createDirectories(uploadPath);
+      }
+      InputStream inputStream = image.getInputStream();
+      Path filePath = uploadPath.resolve(Objects.requireNonNull(image.getOriginalFilename()));
```

Comment:
When saving the image, you should use a unique name in order to avoid images from different sources overwriting each other. For example, you can generate a UUID and use that + the extension of the original filename as the image name. Make sure to store this uuid somewhere so that the image is retrievable.